classes/plugins/DOIPubIdExportPlugin.inc.php

```diff
@@ -0,0 +1,411 @@
+<?php
+
+/**
+ * @file classes/plugins/DOIPubIdExportPlugin.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class DOIPubIdExportPlugin
+ * @ingroup plugins
+ *
+ * @brief Basis class for DOI XML metadata export plugins
+ */
+
+import('lib.pkp.classes.plugins.ImportExportPlugin');
+
+// The status of the DOI.
+define('DOI_EXPORT_STATUS_ANY', '');
+define('DOI_EXPORT_STATUS_NOT_DEPOSITED', 'notDeposited');
+define('DOI_EXPORT_STATUS_MARKEDREGISTERED', 'markedRegistered');
+
+// The status of the DOI.
+define('DOI_EXPORT_ACTION_EXPORT', 'export');
+define('DOI_EXPORT_ACTION_MARKREGISTERED', 'markRegistered');
+define('DOI_EXPORT_ACTION_DEPOSIT', 'deposit');
+define('DOI_EXPORT_ACTION_CHECKSTATUS', 'checkStatus');
+
+// Configuration errors.
+define('DOI_EXPORT_CONFIG_ERROR_DOIPREFIX', 0x01);
+define('DOI_EXPORT_CONFIG_ERROR_SETTINGS', 0x02);
+
+// The name of the setting used to save the registered DOI.
+define('DOI_EXPORT_REGISTERED_DOI', 'registeredDoi');
+
+abstract class DOIPubIdExportPlugin extends ImportExportPlugin {
+	/** @var PubObjectCache */
+	var $_cache;
+
+	function getCache() {
+		if (!is_a($this->_cache, 'PubObjectCache')) {
+			// Instantiate the cache.
+			import('classes.plugins.PubObjectCache');
+			$this->_cache = new PubObjectCache();
+		}
+		return $this->_cache;
+	}
+
+	/**
+	 * Constructor
+	 */
+	function DOIPubIdExportPlugin() {
+		parent::ImportExportPlugin();
+	}
+
+	/**
+	 * @copydoc Plugin::register()
+	 */
+	function register($category, $path) {
+		$success = parent::register($category, $path);
+		if (!Config::getVar('general', 'installed')) return false;
+		if ($success) {
+			HookRegistry::register('AcronPlugin::parseCronTab', array($this, 'callbackParseCronTab'));
+		}
+		$this->addLocaleData();
+		return $success;
+	}
+
+	/**
+	 * @copydoc Plugin::getTemplatePath($inCore)
+	 */
+	function getTemplatePath($inCore = false) {
+		return parent::getTemplatePath($inCore) . 'templates/';
+	}
+
+	/**
+	 * @copydoc Plugin::manage()
+	 */
+	function manage($args, $request) {
+		$user = $request->getUser();
+		$router = $request->getRouter();
+		$context = $router->getContext($request);
+
+		$form = $this->_instantiateSettingsForm($context);
+		$notificationManager = new NotificationManager();
+		switch ($request->getUserVar('verb')) {
+			case 'save':
+				$form->readInputData();
+				if ($form->validate()) {
+					$form->execute();
+					$notificationManager->createTrivialNotification($user->getId(), NOTIFICATION_TYPE_SUCCESS);
+					return new JSONMessage();
+				} else {
+					return new JSONMessage(true, $form->fetch($request));
+				}
+			default:
+				$form->initData();
+				return new JSONMessage(true, $form->fetch($request));
+		}
+		return parent::manage($args, $request);
```

Comment:
I think this is dead code while there's a `default:` clause in place -- but generally it's good to call the parent method if possible. Maybe use something specific instead of `default:`?